Fixes 2 issues related to loading plugins

FinchPowers opened the pull request:
1. Fixes an issue where the error message "Incompatible Cluster: mycluster" would be printed when a plugin fails initialization. Now prints "ERROR - An error occurred while loading plugins: {back trace} Failed to load plugin {xyz} with the following error: {err_msg}. This error may occur if the plugin code is modified after the cluster was created.
2. Fixes an issue where the error message "ERROR - cluster template smallcluster does not exist" when trying to launch a cluster with plugins that fail initialization. Now prints an error message similar than the one from point 1.

Commit:
Fixed an issue of bad errors messages due to plugins

Commit:
Fixed bad call

Commit:
Refined err msg

Commit:
Added debug logging

jtriley commented on starcluster/node.py:
```diff
@@ -4,6 +4,7 @@
 import base64
 import posixpath
 import subprocess
+import traceback
```
No need to import traceback just to log the exception. Use this instead:

```
log.error("Error occured:", exc_info=True)
```

This works for all log levels.

Commit:
Using log instead of traceback

Commit:
Fix refactoring
Using logging instead of traceback.
Moved the getattr call within get_plugins to the existing try and moved the execpt there as well.
